core/geotime/src/main/java/mil/nga/giat/geowave/core/geotime/store/filter/SpatialQueryFilter.java

```diff
@@ -196,8 +204,73 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
+		// cannot intern the prepared geometry. It uses a Object.hashCode().
+		preparedGeometryImage = geometryImageInterner.intern(new GeometryImage(
+				geometryBinary));
+		preparedGeometryImage.init();
```

Comment:
The reason is to delay the construction of the Geometry.  I could have made the 'getter' synchronized, placing the code there.  However, the overhead of synchronizing at that point is unnecessary in inefficient since it is called many times during the filter evaluation.   

After obtaining the image from the interner, there are two possible conditions: the image was already interned by prior code (already initialized) or this new image was interned, requiring initialization.